bigquery_etl/cli/query.py

```diff
@@ -1310,29 +1341,43 @@ def initialize(name, sql_dir, project_id, dry_run):
     for query_file in query_files:
         sql_content = query_file.read_text()
 
-        # Enable init from query.sql files
-        # First deploys the schema, then runs the init
-        # This does not currently verify the accuracy of the schema
+        # Enable initialization from query.sql files
+        # Create the table by deploying the schema and metadata, then run the init.
+        # This does not currently verify the accuracy of the schema or that it
+        # matches the query.
         if "is_init()" in sql_content:
             project = query_file.parent.parent.parent.name
             dataset = query_file.parent.parent.name
             destination_table = query_file.parent.name
-            Schema.from_schema_file(query_file.parent / SCHEMA_FILE).deploy(
-                f"{project}.{dataset}.{destination_table}"
-            )
+            full_table_id = f"{project}.{dataset}.{destination_table}"
+
+            try:
+                client.get_table(full_table_id)
+                raise PreconditionFailed(
+                    f"Table {full_table_id} already exists. The initialization process is terminated."
+                )
+            except NotFound:
+                ctx.invoke(deploy, name=full_table_id, force=True)
+
+            if "@sample_id" in sql_content:
+                sample_ids = list(range(1, 100))
+
             arguments = [
                 "query",
-                "--use_legacy_sql=false",
-                "--replace",
                 "--format=none",
+                "--append_table",
             ]
-            _run_query(
-                query_files=[query_file],
-                project_id=project,
-                public_project_id=None,
-                destination_table=destination_table,
-                dataset_id=dataset,
-                query_arguments=arguments,
+            if dry_run:
+                arguments += ["--dry_run"]
+
+            _initialize_in_parallel(
+                project=project,
+                table=destination_table,
+                dataset=dataset,
+                query_file=query_file,
+                arguments=arguments,
+                parallelism=DEFAULT_PARALLELISM,
+                sample_ids=sample_ids,
```

Comment:
This is going to fail for non-parallelized cases because an empty value for an integer parameter results in an "Unparseable query parameter" error.  You could change the default sample ID value from `""` to `"NULL"`, but IMO it'd be better to simply call `_run_query()` as before for the non-parallelized case.

Reply:
Decided to separate the parallelized an non-parallelized cases for reduced complexity.